django_seed/management/commands/seed.py

```diff
@@ -1,40 +1,105 @@
+        parser.add_argument('-n', '--number', type=int, default=10, help="Number of each model to seed.")
+        parser.add_argument('-e', '--exclude', dest='exclude', action='append', default=[],
+                            help='An app_label or app_label.ModelName to exclude '
+                                 '(use multiple --exclude to exclude multiple apps/models).')
+
+    def handle(self, *app_labels, **options):
+        number = options.get('number')
+        seed_all = options.get('seed_all')
+        exclude = options.get('exclude')
+
+        # Exclude models to skip
+        self.process_models_from_applist(exclude, self.exclude_model)
+
+        # Add models to seed
+        if len(app_labels) == 0:
+            if seed_all:  # Add all models from all apps
+                for app_config in apps.get_app_configs():
+                    for model in app_config.get_models():
+                        self.add_model(model)
+            else:
+                raise SeederCommandError("You must specify a list of apps or the --all flag")
+        else:
+            self.process_models_from_applist(app_labels, self.add_model)
 
         seeder = Seed.seeder()
 
         # don't diplay warnings about non-timezone aware datetimes
         import warnings
+
         warnings.showwarning = lambda *x: None
 
-        for model in app_config.get_models():
+        for model in self.prioritized_models():
             seeder.add_entity(model, number)
             print('Seeding %i %ss' % (number, model.__name__))
 
         pks = seeder.execute()
         print(pks)
 
+    def add_model(self, model):
+        if model not in self.include_models:
+            self.include_models.append(model)
+
+    def exclude_model(self, model):
+        if model not in self.exclude_models:
+            self.exclude_models.append(model)
+
+    def process_models_from_applist(self, applist, func):
+        for app in applist:
+            if '.' in app:  # app.model
+                try:
+                    model = apps.get_model(app)
+                except LookupError:
+                    raise SeederCommandError('Unknown model: %s' % app)
+                else:
+                    func(model)
+            else:  # app
+                try:
+                    app_config = apps.get_app_config(app)
+                except LookupError:
+                    raise SeederCommandError("Unknown application: %s" % app)
+                for model in app_config.get_models():
+                    func(model)
+
+    def prioritized_models(self):
+        dependencies = {}
+        models_to_seed = [m for m in self.include_models if m not in self.exclude_models]
+        for model in models_to_seed:
+            dependencies[model] = set()
+            for field in model._meta.get_fields():
+                # Don't know the exact conditions
+                if field.many_to_one is True and field.concrete and field.blank is False:
```

Comment:
I think this line is the only 1.8 specific code... It's probably better to have it in it's own function anyway. Something like

    def field_has_dependency(self, field)